docs/source/rbac/generate-scope-table.py

```diff
@@ -0,0 +1,99 @@
+import os
+from collections import defaultdict
+
+from pytablewriter import MarkdownTableWriter
+
+from jupyterhub.scopes import scope_definitions
+
+HERE = os.path.abspath(os.path.dirname(__file__))
+
+
+class ScopeTableGenerator:
+    def __init__(self):
+        self.scopes = scope_definitions
+
+    @classmethod
+    def create_writer(cls, table_name, headers, values):
+        writer = MarkdownTableWriter()
+        writer.table_name = table_name
+        writer.headers = headers
+        writer.value_matrix = values
+        writer.margin = 1
+        return writer
+
+    def _get_scope_relationships(self):
```

Comment:
I'm not sure this is an issue, but this functions generates duplicates, e.g. `('read:users', 'read:users:activity')` and `('users:activity', 'read:users:activity')`, which can be seen in the generated table:

![image](https://user-images.githubusercontent.com/4711805/120980890-5b327080-c777-11eb-9b5c-4295a5be37be.png)

Reply:
@davidbrochart the duplication here is intentional, it is to show that the `read:users:activity` is a sub-scope of both `read:users` and `users:activity`, which are not themselves related. Is it confusing?

Reply:
> Is it confusing?

Maybe it could help to mention that this special scope can be specified using two different syntaxes, since I believe it is the only one?

Reply:
We will definitely mention this, there is another similar duplicate with `read:users:name` which is a sub-scope of both `read:users` and `read:users:servers` (again, not related). We will highlight the duplications and the reasons behind them in the docs, they are still being updated as we speak. Thank you for the inquiry and suggestion!